ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,576 @@
+   FROM campaigns
+  WHERE company_id = 5
+  ORDER BY monthly_budget DESC
+  LIMIT 10;
+
+The EXPLAIN output shows that Citus routes this to a single worker node, inside which PostgreSQL uses its usual tactics.
+
+::
+
+  ┌────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+  │                                                 QUERY PLAN                                                 │
+  ├────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
+  │ Custom Scan (Citus Router)  (cost=0.00..0.00 rows=0 width=0)                                               │
+  │   Task Count: 1                                                                                            │
+  │   Tasks Shown: All                                                                                         │
+  │   ->  Task                                                                                                 │
+  │         Node: host=ec2-34-224-105-231.compute-1.amazonaws.com port=5432 dbname=citus                       │
+  │         ->  Limit  (cost=9.51..9.52 rows=2 width=104)                                                      │
+  │               ->  Sort  (cost=9.51..9.52 rows=2 width=104)                                                 │
+  │                     Sort Key: monthly_budget DESC                                                          │
+  │                     ->  Bitmap Heap Scan on campaigns_102046 campaigns  (cost=4.16..9.50 rows=2 width=104) │
+  │                           Recheck Cond: (company_id = 5)                                                   │
+  │                           ->  Bitmap Index Scan on campaigns_pkey_102046  (cost=0.00..4.16 rows=2 width=0) │
+  │                                 Index Cond: (company_id = 5)                                               │
+  └────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+
+Updates work too. Let's double the budget for all campaigns.
+
+::
+
+  UPDATE campaigns
+  SET monthly_budget = monthly_budget*2
+  WHERE company_id = 5;
+
+In all these queries, the filter routes SQL execution directly inside a worker. Full SQL support is available once queries are "pushed down" to a worker like this. When people scale applications with NoSQL databases they suffer the lack of transactions and joins. However per-tenant transactions work fine in Citus:
+
+::
+
+  -- transactionally remove campaign 46 and all its ads
+
+  BEGIN;
+  DELETE from campaigns where id = 46 AND company_id = 5;
+  DELETE from ads where campaign_id = 46 AND company_id = 5;
+  COMMIT;
+
+As a final demonstration of SQL support, this query includes a joins, grouping, aggregation and a window function, and it executes fine. The query ranks the ads in each campaign by the count of their impressions.
+
+::
+
+  SELECT a.campaign_id,
```

Comment:
This is a good start, I'll spend a few minutes today evening or tom morning trying to see if I can come up with something better.